zeebe/atomix/cluster/src/test/java/io/atomix/raft/roles/RaftFollowerFlushErrorTest.java

```diff
@@ -71,11 +72,15 @@ public void shouldAppendEntryOnAllNodesWhenFollowerFailsFlush() throws Throwable
 
     // all members are still registered
     assertThat(raftRule.getMemberLogs().size()).isEqualTo(MEMBERS);
-    // all members are either LEADER or FOLLOWER
-    assertThat(
-            raftRule.getServers().stream()
-                .filter(s -> s.getRole() == Role.FOLLOWER || s.getRole() == Role.LEADER)
-                .count())
-        .isEqualTo(MEMBERS);
+    Awaitility.await("Until all members are FOLLOWER or LEADER")
+        .untilAsserted(
+            () -> {
+              // all members are either LEADER or FOLLOWER
+              final var roles = raftRule.getServers().stream().map(RaftServer::getRole).toList();
+              assertThat(roles.stream().filter(r -> r == Role.FOLLOWER || r == Role.LEADER).count())
```

Comment:
:wrench: More descriptively, you could use `assertThat(roles).allMatch` or `assertThat(roles).allSatisfy`. Especially `allSatisfy` will give you better error messages.